Structure_Engine/Query/IsNull.cs

```diff
@@ -0,0 +1,460 @@
+        [Output("isNull", "True if the Node or its defining properties are null.")]
+        public static bool IsNull(this Node node, [CallerMemberName] string methodName = "Method", string msg = "")
+        {
+            // Check Node and Position
+            if (node == null)
+            {
+                ErrorMessage(methodName, "Node", msg);
+                return true;
+            }
+            else if (node.Position == null)
+            {
+                ErrorMessage(methodName, "Node Position", msg);
+                return true;
+            }
+
+            return false;
+        }
+
+        /***************************************************/
+
+        [PreviousVersion("4.2", "BH.Engine.Structure.Compute.NullCheck(BH.oM.Structure.Elements.Bar, System.String)")]
+        [Description("Checks if a Bar or its defining properties are null and outputs relevant error message.")]
+        [Input("bar", "The Bar to test for null.")]
+        [Input("methodName", "The name of the method to reference in the error message.")]
+        [Input("msg", "Optional message to be returned in addition to the generated error message.")]
+        [Output("isNull", "True if the Bar or its defining properties are null.")]
+        public static bool IsNull(this Bar bar, [CallerMemberName] string methodName = "Method", string msg = "")
+        {
+            // Check bar
+            if (bar == null)
+            {
+                ErrorMessage(methodName, "Bar", msg);
+                return true;
+            }
+            else if (bar.StartNode.IsNull(methodName, "The Node (StartNode) is owned by a Bar.") || bar.EndNode.IsNull(methodName, "The Node (EndNode) is owned by a Bar."))
+                return true;
+
+            return false;
+        }
+
+        [PreviousVersion("4.2", "BH.Engine.Structure.Compute.NullCheck(BH.oM.Structure.Elements.FEMesh, System.String, System.Boolean, System.Boolean, System.Collections.Generic.List<System.Int32>)")]
+        [Description("Checks if an FEMesh or its defining properties are null and outputs relevant error message.")]
+        [Input("mesh", "The FEMesh to test for null.")]
+        [Input("methodName", "Optional name of the method to reference in the error message.")]
+        [Input("checkFaces", "Optional bool to tell the method whether to check FEMeshFaces or not.")]
+        [Input("checkNodes", "Optional bool to tell the method whether to check mesh Nodes or not.")]
+        [Input("nodeListIndices", "Optional list of nodes to limit check to.")]
+        [Input("msg", "Optional message to be returned in addition to the generated error message.")]
+        [Output("isNull", "True if the FEMesh or its defining properties are null.")]
+        public static bool IsNull(this FEMesh mesh, [CallerMemberName] string methodName = "Method", bool checkFaces = true, bool checkNodes = true, List<int> nodeListIndices = null, string msg = "")
```

Comment:
As mentioned in above comment. Scope to put the `methodName` last in the argument chain, even after the additional `msg`, to all other parameters can be changed, while still always keeping the benefit of the `[CallerMemberName]` attribute.

Would mean some additional refactoring, but might be worth it..

If change is made, need to trickle down to all the methods in the class, and would if so also change the Geometry and Spatial methods.